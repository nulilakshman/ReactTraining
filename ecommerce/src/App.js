import logo from './logo.svg';
import './App.css';
import { Link, Outlet } from 'react-router-dom';
import Header from './Components/Header'
import Footer from './Components/Footer'
function App() {
  return (
    <div id="layout-wrapper">
      <Header />
      <Outlet></Outlet>
      <Footer />
    </div>
  );
}

export default App;
