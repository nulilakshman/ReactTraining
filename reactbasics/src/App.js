import logo from './logo.svg';
import './App.css';
import { useState } from 'react';

function App() {
  const [counter, setCounter] = useState(0)
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Welcome to React !!! Content changed !!!

        </p>
        <button data-testid="btnIncrement" onClick={() => setCounter(counter + 1)}  >Add Counter</button>

        Counter is :  <p data-testid='countervalue'>{counter}</p>
      </header>
    </div>
  );
}

export default App;
