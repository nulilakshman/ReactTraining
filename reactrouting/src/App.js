import logo from './logo.svg';
import './App.css';
import { Link, Outlet } from 'react-router-dom';
import Footer from './Footer';

function App() {
  return (
    <div className="App">
      <div>
        <h1>React Learnings</h1>
      </div>
      <Link to='/home'>Home</Link> |
      <Link to='/about'>About Us</Link> |
      <Link to='/about/10'>About Us With value</Link> |
      <Link to='/about/username/testuser/uid/45'>About Us With name</Link> |
      <Link to='/users'>Users</Link> |
      <Link to='/contactus'>Contact Us</Link> |
      <Link to='/contactus-1'>Error</Link>

      <Outlet></Outlet>

      <div>
        <Footer></Footer>
      </div>
    </div>
  );
}

export default App;
