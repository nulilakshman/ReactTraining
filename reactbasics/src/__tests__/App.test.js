import React from "react";
import App from '../App';
import { render, screen, cleanup, fireEvent } from '@testing-library/react'

afterEach(cleanup)

describe("Tests related to App Component", () => {

    let component;
    beforeEach(() => {
        component = render(<App></App>);
    })

    it("should load snapshot", () => {
        expect(component).toMatchSnapshot();
    })

    it("should find welcome note", () => {
        const text = screen.getByText(/Welcome to React/i);
        expect(text).toBeInTheDocument();
    })

    it("should increment counter on button click", () => {
        fireEvent.click(component.getByTestId('btnIncrement'));
        fireEvent.click(component.getByTestId('btnIncrement'));
        expect(component).toMatchSnapshot();
        const value = component.getByTestId('countervalue')
        expect(value).toHaveTextContent(2)
    
    })


});