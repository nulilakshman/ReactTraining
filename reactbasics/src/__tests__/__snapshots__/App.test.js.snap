// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests related to App Component should increment counter on button click 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="App-header"
        >
          <img
            alt="logo"
            class="App-logo"
            src="logo.svg"
          />
          <p>
            Welcome to React !!! Content changed !!!
          </p>
          <button
            data-testid="btnIncrement"
          >
            Add Counter
          </button>
          Counter is :  
          <p
            data-testid="countervalue"
          >
            2
          </p>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <img
          alt="logo"
          class="App-logo"
          src="logo.svg"
        />
        <p>
          Welcome to React !!! Content changed !!!
        </p>
        <button
          data-testid="btnIncrement"
        >
          Add Counter
        </button>
        Counter is :  
        <p
          data-testid="countervalue"
        >
          2
        </p>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Tests related to App Component should load snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <header
          class="App-header"
        >
          <img
            alt="logo"
            class="App-logo"
            src="logo.svg"
          />
          <p>
            Welcome to React !!! Content changed !!!
          </p>
          <button
            data-testid="btnIncrement"
          >
            Add Counter
          </button>
          Counter is :  
          <p
            data-testid="countervalue"
          >
            0
          </p>
        </header>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <header
        class="App-header"
      >
        <img
          alt="logo"
          class="App-logo"
          src="logo.svg"
        />
        <p>
          Welcome to React !!! Content changed !!!
        </p>
        <button
          data-testid="btnIncrement"
        >
          Add Counter
        </button>
        Counter is :  
        <p
          data-testid="countervalue"
        >
          0
        </p>
      </header>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
